test/Anoma/Compilation/Positive.hs

```diff
@@ -0,0 +1,522 @@
+module Anoma.Compilation.Positive where
+
+import Base
+import Juvix.Compiler.Backend (Target (TargetAnoma))
+import Juvix.Compiler.Nockma.Evaluator
+import Juvix.Compiler.Nockma.Language
+import Juvix.Compiler.Nockma.Translation.FromSource.QQ
+import Juvix.Compiler.Nockma.Translation.FromTree
+import Juvix.Prelude qualified as Prelude
+import Nockma.Base
+import Nockma.Eval.Positive
+
+root :: Prelude.Path Abs Dir
+root = relToProject $(mkRelDir "tests/Anoma/Compilation/positive")
+
+mkAnomaCallTest' :: Bool -> Text -> Prelude.Path Rel Dir -> Prelude.Path Rel File -> [Term Natural] -> Check () -> TestTree
+mkAnomaCallTest' enableDebug _testName relRoot mainFile args _testCheck =
+  testCase (unpack _testName) (mkTestIO >>= mkNockmaAssertion)
+  where
+    mkTestIO :: IO Test
+    mkTestIO = do
+      _testProgramSubject <- withRootCopy compileMain
+      let _testProgramFormula = anomaCall args
+          _testEvalOptions = defaultEvalOptions
+      return Test {..}
+
+    withRootCopy :: (Prelude.Path Abs Dir -> IO a) -> IO a
+    withRootCopy action = withSystemTempDir "test" $ \tmpRootDir -> do
```

Comment:
The stored standard library is not shared between different tests, which defeats per-module compilation and makes the tests run 3-4 times slower than they otherwise would.

Reply:
Running the tests concurrently in the same directory triggers the leaking IO bypassing the TaggedLock that we've seen before. It's possible that using Effectful will fix this. An alternative would be to synchronously build the standard library somehow before running the test suite and copy that to each temporary location.

Reply:
Somehow, it works for the Juvix->Native/Wasm compilation tests (in `tests/Compilation/positive`). What's different?

I remember I've been dealing specifically with this issue when implementing per-module-compilation, and ultimately used file-level locking which solved the problem.

Reply:
I don't remember what's happening exactly for the main exec pipeline compilation tests, but the stored standard library modules seem to be reused judging by decrease in runtime after first test.

Reply:
The anoma compilation tests are using the same functions to run the pipeline as the main exec compilation tests, yet they fail for the leaking IO / tagged lock issue. TaggedLock in general is too unreliable and I'm thinking about a plan to remove it. I propose to leave the anoma compilation tests as they are (using separate directories) for now until we work out how we will replace tagged lock.